sdk/core/src/http_client.rs

```diff
@@ -97,15 +100,15 @@ impl HttpClient for hyper::Client<HttpsConnector<hyper::client::HttpConnector>>
 impl HttpClient for reqwest::Client {
     async fn execute_request(
         &self,
-        request: Request<&[u8]>,
+        request: Request<Bytes>,
     ) -> Result<Response<Vec<u8>>, Box<dyn std::error::Error + Sync + Send>> {
         let mut reqwest_request =
             self.request(request.method().clone(), &request.uri().to_string());
         for header in request.headers() {
             reqwest_request = reqwest_request.header(header.0, header.1);
         }
 
-        let body = String::from_utf8(request.body().to_vec())?;
```

Comment:
The problem here is that byte array reference gets cloned with `to_vec` & then converts it to a string, only to have the string converted to bytes later. It is more efficient to own the bytes and send them without the conversion.

Reply:
Yes you are right, the string conversion is not really necessary.
Also the idea to use ref counting here (by using `bytes`) is really good. 👍 👍